src/NHibernate/Dialect/Dialect.cs

```diff
@@ -723,15 +724,21 @@ public virtual string GetDropForeignKeyConstraintString(string constraintName)
 			return " drop constraint " + constraintName;
 		}
 
+
 		/// <summary>
 		/// The syntax that is used to check if a constraint does not exists before creating it
 		/// </summary>
 		/// <param name="table">The table.</param>
 		/// <param name="name">The name.</param>
 		/// <returns></returns>
+		[Obsolete("Can cause issues when a custom schema is defined(https://nhibernate.jira.com/browse/NH-1285). The new overload with the defaultSchema parameter should be used instead")]
 		public virtual string GetIfNotExistsCreateConstraint(Table table, string name)
 		{
-			return "";
+			var catalog = table.Catalog == null ? null : table.GetQuotedCatalog(this);
+			var schema = table.Schema == null ? null : table.GetQuotedSchema(this);
+			var tableName = table.GetQuotedName(this);
+
+			return GetIfNotExistsCreateConstraint(catalog, schema, tableName, name);
```

Comment:
Since the called method is public too, it will have to check if handling quoting is required too, for cases where it is called by some other code. So quoting here seems redundant. Same for next methods.